diff --git a/include/depends.mk b/include/depends.mk
index 65e9b25..828ceb8 100644
--- a/include/depends.mk
+++ b/include/depends.mk
@@ -13,7 +13,7 @@
 
 DEP_FINDPARAMS := -x "*/.svn*" -x ".*" -x "*:*" -x "*\!*" -x "* *" -x "*\\\#*" -x "*/.*_check"
 
-find_md5=find $(1) -type f $(patsubst -x,-and -not -path,$(DEP_FINDPARAMS) $(2)) | md5s
+find_md5=$(SH_FUNC) find $(1) -type f $(patsubst -x,-and -not -path,$(DEP_FINDPARAMS) $(2)) | md5s
 
 define rdep
   .PRECIOUS: $(2)
diff --git a/rules.mk b/rules.mk
index 990790b..17a629e 100644
--- a/rules.mk
+++ b/rules.mk
@@ -152,6 +152,7 @@ endif
 export PATH:=$(TARGET_PATH)
 export STAGING_DIR
 export GCC_HONOUR_COPTS:=0
+export SH_FUNC:=. $(INCLUDE_DIR)/shell.sh; 
 
 PKG_CONFIG:=$(STAGING_DIR_HOST)/bin/pkg-config
 
diff --git a/include/kernel.mk b/include/kernel.mk
index 63df959..6290d9b 100644
--- a/include/kernel.mk
+++ b/include/kernel.mk
@@ -68,6 +68,7 @@ define KernelPackage/Defaults
 endef
 
 define ModuleAutoLoad
+	$(SH_FUNC) \
 	export modules=; \
 	add_module() { \
 		mkdir -p $(2)/etc/modules.d; \
diff --git a/include/package-ipkg.mk b/include/package-ipkg.mk
index 0dd398e..9e53cbe 100644
--- a/include/package-ipkg.mk
+++ b/include/package-ipkg.mk
@@ -94,6 +94,8 @@ ifeq ($(DUMP),)
 	echo "Package: $(1)" > $$(IDIR_$(1))/CONTROL/control
 	echo "Version: $(VERSION)" >> $$(IDIR_$(1))/CONTROL/control
 	( \
+		echo "Package: $(1)"; \
+		echo "Version: $(VERSION)"; \
 		DEPENDS='$(EXTRA_DEPENDS)'; \
 		for depend in $$(filter-out @%,$$(IDEPEND_$(1))); do \
 			DEPENDS=$$$${DEPENDS:+$$$$DEPENDS, }$$$${depend##+}; \
@@ -108,10 +110,10 @@ ifeq ($(DUMP),)
 		echo "Maintainer: $(MAINTAINER)"; \
 		echo "Architecture: $(PKGARCH)"; \
 		echo "Installed-Size: 0"; \
-		echo -n "Description: "; getvar $(call shvar,Package/$(1)/description) | sed -e 's,^[[:space:]]*, ,g'; \
- 	) >> $$(IDIR_$(1))/CONTROL/control
+		echo -n "Description: "; $(SH_FUNC) getvar $(call shvar,Package/$(1)/description) | sed -e 's,^[[:space:]]*, ,g'; \
+ 	) > $$(IDIR_$(1))/CONTROL/control
 	chmod 644 $$(IDIR_$(1))/CONTROL/control
-	(cd $$(IDIR_$(1))/CONTROL; \
+	$(SH_FUNC) (cd $$(IDIR_$(1))/CONTROL; \
 		$($(1)_COMMANDS) \
 	)
 	$(call Package/$(1)/install,$$(IDIR_$(1)))
diff --git a/include/target.mk b/include/target.mk
index 0db3815..f5e4609 100644
--- a/include/target.mk
+++ b/include/target.mk
@@ -67,10 +67,10 @@ define Profile
 		echo "Target-Profile-Kconfig: yes"; \
 	fi; \
 	echo "Target-Profile-Config: "; \
-	getvar "$(call shvar,Profile/$(1)/Config)"; \
+	$(SH_FUNC) getvar "$(call shvar,Profile/$(1)/Config)"; \
 	echo "@@"; \
 	echo "Target-Profile-Description:"; \
-	getvar "$(call shvar,Profile/$(1)/Description)"; \
+	$(SH_FUNC) getvar "$(call shvar,Profile/$(1)/Description)"; \
 	echo "@@"; \
 	echo;
   ifeq ($(CONFIG_TARGET_$(call target_conf,$(BOARD)_$(if $(SUBTARGET),$(SUBTARGET)_)$(1))),y)
@@ -186,7 +186,7 @@ define BuildTargets/DumpCurrent
 	 echo 'Linux-Release: $(LINUX_RELEASE)'; \
 	 echo 'Linux-Kernel-Arch: $(LINUX_KARCH)'; \
 	 echo 'Target-Description:'; \
-	 getvar $(call shvar,Target/Description); \
+	 $(SH_FUNC) getvar $(call shvar,Target/Description); \
 	 echo '@@'; \
 	 echo 'Default-Packages: $(DEFAULT_PACKAGES)'; \
 	 $(DUMPINFO)
diff --git a/include/unpack.mk b/include/unpack.mk
index 0151675..2d35863 100644
--- a/include/unpack.mk
+++ b/include/unpack.mk
@@ -71,10 +71,10 @@ ifeq ($(strip $(UNPACK_CMD)),)
 endif
 
 ifdef PKG_BUILD_DIR
-  PKG_UNPACK ?= $(call UNPACK_CMD,$(PKG_BUILD_DIR)) $(call CRLF_CMD,$(PKG_BUILD_DIR))
+  PKG_UNPACK ?= $(SH_FUNC) $(call UNPACK_CMD,$(PKG_BUILD_DIR)) $(call CRLF_CMD,$(PKG_BUILD_DIR))
 endif
 ifdef HOST_BUILD_DIR
-  HOST_UNPACK ?= $(call UNPACK_CMD,$(HOST_BUILD_DIR)) $(call CRLF_CMD,$(HOST_BUILD_DIR))
+  HOST_UNPACK ?= $(SH_FUNC) $(call UNPACK_CMD,$(HOST_BUILD_DIR)) $(call CRLF_CMD,$(HOST_BUILD_DIR))
 endif
 
 endif # PKG_SOURCE
