diff --git a/toolchain/gcc/Makefile b/toolchain/gcc/Makefile
index ba0feb7..c6e842a 100644
--- a/toolchain/gcc/Makefile
+++ b/toolchain/gcc/Makefile
@@ -172,6 +172,7 @@ define Stage0/Configure
 endef
 
 define Stage0/Compile
+	echo "MAKEINFO = :" >> $(HOST_BUILD_DIR0)/Makefile
 	$(GCC_MAKE) -C $(HOST_BUILD_DIR0) all-gcc
 endef
 
@@ -188,6 +189,7 @@ define Stage1/Configure
 endef
 
 define Stage1/Compile
+	echo "MAKEINFO = :" >> $(HOST_BUILD_DIR1)/Makefile
 	$(CP) $(BUILD_DIR_TOOLCHAIN)/linux-dev/* $(BUILD_DIR_TOOLCHAIN)/$(LIBC)-dev/
 	$(GCC_MAKE) -C $(HOST_BUILD_DIR1) \
 		all-build-libiberty \
@@ -222,6 +224,7 @@ define Stage2/Configure
 endef
 
 define Stage2/Compile
+	echo "MAKEINFO = :" >> $(HOST_BUILD_DIR2)/Makefile
 	$(GCC_MAKE) -C $(HOST_BUILD_DIR2) all
 endef
 
diff --git a/toolchain/binutils/Makefile b/toolchain/binutils/Makefile
index a3ce789..e9d22e2 100644
--- a/toolchain/binutils/Makefile
+++ b/toolchain/binutils/Makefile
@@ -72,6 +72,7 @@ define Host/Configure
 endef
 
 define Host/Compile
+	echo "MAKEINFO = :" >> $(HOST_BUILD_DIR)/Makefile
 	$(MAKE) -C $(HOST_BUILD_DIR) all
 endef
 
