diff --git a/toolchain/gcc/Makefile b/toolchain/gcc/Makefile
index ba0feb7..c6e842a 100644
--- a/toolchain/gcc/Makefile
+++ b/toolchain/gcc/Makefile
@@ -83,7 +83,7 @@ GCC_CONFIGURE:= \
 		--disable-nls \
 		$(if $(CONFIG_GCC_USE_GRAPHITE),--with-host-libstdcxx=-lstdc++) \
 		$(SOFT_FLOAT_CONFIG_OPTION) \
-		$(call qstrip,$(CONFIG_EXTRA_GCC_CONFIG_OPTIONS)) \
+		$(call qstrip,$(CONFIG_EXTRA_GCC_CONFIG_OPTIONS) CFLAGS=-fgnu89-inline CXXFLAGS=-fgnu89-inline) \
 		$(if $(CONFIG_GCC_VERSION_4_3_3_CS),--enable-poison-system-directories)
 
 ifneq ($(CONFIG_GCC_VERSION_4_3)$(CONFIG_GCC_VERSION_4_4),)
